[Security Solution] EUI Refresh Followup Pt. 1

rylnd opened the pull request:
## Summary

This is the first batch of work involving https://github.com/elastic/kibana/issues/202052. It includes the following tasks:

- [x] (Important) Review usage of "success" colors for positive, semantic intent.([details](https://github.com/elastic/kibana/issues/202052#success)).
- [x] (Important) Review and update color palette usage. ([details](#color-palettes))
- [x] (Important) Replace vis colors used outside of visualizations with color tokens  . ([details](https://github.com/elastic/kibana/issues/202052#viz-colors))
- [x] (Important) Replace custom colors with semantic tokens ([details](https://github.com/elastic/kibana/issues/202052#custom-colors))
- [x] (Important) Replace color calculation functions with semantic tokens. ([details](https://github.com/elastic/kibana/issues/202052#color-functions))
- [x] Update tokens to use the new naming scheme. ([details](https://github.com/elastic/kibana/issues/202052#renamed))

### What changed?
Broadly, these changes involve two main areas: Exception Lists and the Rule Creation form (specifically: the EQL query bar and the "time" inputs (Suppression Duration, Interval, Lookback)).

I also found an unused component, and fixed a minor UI bug with Exception Comments.

Screenshots (before and after, and with both themes) are included below.

Also: as a majority of these components' usage of theme variables is done with `@styled-components`, and we are transitioning away from that toward `@emotion`, I also opted to remove usages of `@styled-components` wherever possible, as that change was negligible in comparison with the theming changes. If a file being changed included references to static eui theme variables, e.g. `euiThemeVars`, those were removed as well (but many are still outstanding).


### Screenshots

__NOTE__: Most of the "Before" screenshots were taken from a cluster using the Borealis theme.

<details>
  <summary>
    <h3>Rule Exceptions Tab</h3>
  </summary>
  <kbd>
    <h3>Before</h3>
<img width="1594" alt="Rule Exceptions - Before" src="https://github.com/user-attachments/assets/56feb5e8-b5ec-4bdc-9372-968056e6bca3" />

  </kbd>
  <kbd>
    <h3>After (Amsterdam)</h3>
<img width="805" alt="Rule Exceptions - After (Amsterdam)" src="https://github.com/user-attachments/assets/a6ab3d47-acc7-4eb6-8a0c-bd9ccafc2021" />

  </kbd>
  <kbd>
    <h3>After (Borealis)</h3>
<img width="805" alt="Rule Exceptions - After (Borealis)" src="https://github.com/user-attachments/assets/9fbd8a17-0afb-4ccc-95ea-5bab4da5b254" />

  </kbd>
</details>

<details>
  <summary>
    <h3>Shared Exception Lists</h3>
  </summary>
  <kbd>
    <h3>Before</h3>
<img width="1594" alt="Shared Exception Lists - Before" src="https://github.com/user-attachments/assets/b2a69f9c-4892-4732-a2e1-e167461680d1" />

  </kbd>
  <kbd>
    <h3>After (Amsterdam)</h3>
<img width="805" alt="Shared Exception Lists - After (Amsterdam)" src="https://github.com/user-attachments/assets/1aea0dff-97ee-4482-9a66-38c4be2baea8" />

  </kbd>
  <kbd>
    <h3>After (Borealis)</h3>
<img width="805" alt="Shared Exception Lists - After (Borealis)" src="https://github.com/user-attachments/assets/76be9446-17fb-451d-b15a-99166cb1f588" />
  </kbd>
</details>

<details>


  <summary>
    <h3>Shared List Details</h3>
  </summary>
  <kbd>
    <h3>Before</h3>
<img width="1594" alt="Shared List Details - Before" src="https://github.com/user-attachments/assets/5eeec91d-ea96-4e3c-8c7d-b5e6bb8bf35a" />

  </kbd>
  <kbd>
    <h3>After (Amsterdam)</h3>
<img width="805" alt="Shared List Details - After (Amsterdam)" src="https://github.com/user-attachments/assets/5229b850-356c-40e7-80f1-d32156f97a56" />

  </kbd>
  <kbd>
    <h3>After (Borealis)</h3>
<img width="805" alt="Shared List Details - After (Borealis)" src="https://github.com/user-attachments/assets/81219613-c329-48f8-8fd3-857207d0ada0" />

  </kbd>
</details>

<details>
  <summary>
    <h3>EQL Query Input</h3>
  </summary>
  <kbd>
    <h3>Before</h3>
<img width="1085" alt="EQL Bar - Before" src="https://github.com/user-attachments/assets/b19cf18b-a5d2-4001-bebc-9154af8e9b43" />

  </kbd>
  <kbd>
    <h3>After (Amsterdam)</h3>
<img width="1085" alt="EQL Bar - After (Amsterdam)" src="https://github.com/user-attachments/assets/70911ad2-9e52-4b65-b305-cd90a6be1fd2" />

  </kbd>
  <kbd>
    <h3>After (Borealis)</h3>
<img width="1085" alt="EQL Bar - After (Borealis)" src="https://github.com/user-attachments/assets/92a789d0-fffa-483c-bcb2-fc02b5e8ce10" />

  </kbd>
</details>


<details>
  <summary>
    <h3>Suppression Fields</h3>
  </summary>
  <kbd>
    <h3>Before</h3>
<img width="1085" alt="Suppression Fields - Before" src="https://github.com/user-attachments/assets/458c3464-0eba-47d4-abbe-8de0b333b8ec" />

  </kbd>
  <kbd>
    <h3>After (Amsterdam)</h3>
<img width="1085" alt="Suppression Fields - After (Amsterdam)" src="https://github.com/user-attachments/assets/b6fb67b0-7908-4fe8-9e02-d9e14c8f97c5" />

  </kbd>
  <kbd>
    <h3>After (Borealis)</h3>
<img width="1085" alt="Suppression Fields - After (Borealis)" src="https://github.com/user-attachments/assets/f48c24a5-50ea-41fd-af45-b3d32f20921b" />

  </kbd>

</details>

<details>
  <summary>
    <h3>Suppression Fields (Disabled)</h3>
  </summary>
  <kbd>
    <h3>Before</h3>
<img width="1085" alt="Suppression Fields (Disabled) - Before" src="https://github.com/user-attachments/assets/e60aa561-d8ce-4b23-8694-e6443ad37c16" />

  </kbd>
  <kbd>
    <h3>After (Amsterdam)</h3>
<img width="1085" alt="Suppression Fields (Disabled) - After (Amsterdam)" src="https://github.com/user-attachments/assets/45277962-482d-484c-b8bd-3a587b7c16d1" />

  </kbd>
  <kbd>
    <h3>After (Borealis)</h3>
<img width="1085" alt="Suppression Fields (Disabled) - After (Borealis)" src="https://github.com/user-attachments/assets/69f6217b-a2c2-4c6a-be9e-80dfd0a4e5f1" />

  </kbd>
</details>

<details>
  <summary>
    <h3>Exception Comments UI Bug</h3>
  </summary>
  <kbd>
    <h3>Before</h3>
<img width="805" alt="Comments UI Bug - Before" src="https://github.com/user-attachments/assets/652d35fa-8ff2-4850-a9a4-79cec3b4b698" />

  </kbd>
  <kbd>
    <h3>After</h3>
<img width="805" alt="Comments UI Bug - After" src="https://github.com/user-attachments/assets/a8cc1cd6-b25f-4694-a479-c35f45eae6aa" />

  </kbd>
</details>

### Checklist

- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

rylnd committed:
Replace usage of `warningText` colors within DE and RM code

`warningText -> textWarning`. This also replaces some uses of the
now-deprecated `euiThemeVars` with use of the `useEuiTheme` hook.

rylnd committed:
Move EQL textarea styles to @emotion styles

* This uses the theme to set a min-height on the textarea, since it is a
  best practice with a resizable element. The dropped styles were no
  longer necessary with the EUI component.

rylnd committed:
Fix style of Suppression duration input when disabled

Both box-shadow and border were being applied here, which combined with
the border-radius lead to a weird-looking input. This removes the
box-shadow.

rylnd committed:
Replace styled-components with @emotion in EuiQueryBar

* Removes uses of deprecated tokens
* Aligns EqlQueryBar closer to a normal form component (no custom label)

rylnd committed:
Replace usage of euiThemeVars in Exception Item Comments

These were "solved" by using the `useEuiTheme` hook and placing the
generation of styles within the component itself, as opposed to
statically outside them.

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!

rylnd committed:
WIP: Attempting to use styles in non-component functions

In the case of `generateLinkedRulesMenuItems`, the function was a react
component in everything but name. I renamed it so that the react-hooks
linter rules wouldn't complain, but I left the manual invocations of the
element since they're fine.

rylnd committed:
Revert "WIP: Attempting to use styles in non-component functions"

This reverts commit 0489ee1f806517f80c08ef622ec9b81ac82fba9d.

This ends up not working since that component now calls a hook, and you
can't call a hook from e.g. the `useMemo` that wraps the
link-generating, in at least one instance. I'll have to come back to
this one.

rylnd committed:
Replace direct hex color usage in Exception Item Conditions component(s)

This replaces the use hex colors (to style a border), swapping instead
for the now-preferred euiTheme context, as well as using `@emotion` when possible.

These changes are applied here to both the in-plugin and packaged
versions of this component, which are used in two different sections. It
also attempts to bring them more inline with one another by applying a
missing `eui-xScroll` class that did not exist on the plugin variant of
this component.

rylnd committed:
Fix/limit scoping of accordion styles

These styles (particularly the box-shadow) were being incorrectly
applied to all child elements, which caused the inner Comments accordion
to be styled incorrectly. I failed to catch this when it was introduced
in #178023.

rylnd committed:
Fix/simplify styles on Exception List Comments

This fixes the styling of comments, namely on the accordion, by:

1. Ensuring consistency between the two implementations
2. Removing unnecessary styles

rylnd committed:
Remove references to hex colors/EuiThemeVars in more exception components

The MetaInfo component was easy to find, but it took a while to track
down where the MenuItems component is used: on the exception DETAILS
page, which is only accessed through the shared exception list page by
clicking on an individual list.

rylnd committed:
Swap use of hex color reference with semantic equivalent

Finally, an easy one!

rylnd committed:
Remove euiThemeVars, hex color, and styled-components from TitleBadge

This component is used on the shared exceptions page, where exceptions
lists are listed.

rylnd committed:
Remove usage of transparentize()

We only use this in our Duration components, as they are custom. It's
worth noting that we've re-implemented this in three different places,
as you can see here, but we can deduplicate later. These are EuiSelects
where Seconds/Minutes/Hours are chosen for either rule scheduling or
suppression duration.

Instead of using this custom color, it's been replaced with
`backgroundBasePrimary`, making it more in light with other similar form
controls.

I also replaced some usages of styled-components, here.

rylnd committed:
Remove custom styling of labelAppend on ScheduleItem

In testing this CSS (to determine a replacement), I discovered that it's
adding a fixed margin-left to labelAppend, which may extend outside the
bounds of the form row. The margin size itself seemed arbitrary. The
container for both labels is also a fixed size.

Instead, I opted to change the containing flexbox to justify as
`spaceBetween`, maximizing the distance between the items. If the
incoming `labelAppend` prop needs to add additional styling, the caller
can add that (although in testing I did not see usage of this prop).

rylnd committed:
Remove unused form component and associations

This was replaced by `ScheduleRuleField` in #200304. I discovered
this while looking for usages of EUI color helpers, but upon further
inspection it can just be pruned.

rylnd committed:
More accurate variable names

I was reusing style names from the similar Suppression Duration
components, but that's not correct, here. These apply to the Scheduling
fields during rule creation, so "duration" isn't accurate.

rylnd committed:
Remove list-header styles file

This contained now-outdated references to EuiThemeVars that needed to be
inlined to the consuming component (in order to take advantage of the
euiTheme context/hook).

On the ListHeader component, I was able to remove a negative margin by
using `bottomBorder="extended"` as per the documentation. However, due
to the way the content is currently constructed, we still need a
negative margin-top in order to offset a hardcoded EuiSpacer in the
EuiPageHeader component (for which a comment was included on the style).

I was not able to use the full `useEuiFontSize` helper as the
line-heights have changed significantly from the original layout, hence
the `font-size: ().fontSize` pattern.

rylnd committed:
Convert ExceptionItemComments styles to @emotion

For reasons I'm not yet clear on, the template literal syntax does not
work in this file, but the object notation does.

rylnd committed:
Add space between existing comments and the "add comment" textarea

This was a minor UI bug that only shows when you're editing an exception
item in order to add another comment.

elasticmachine commented:
Pinging @elastic/security-detection-engine (Team:Detection Engine)

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/styled_components_mapping'

jbudz reviewed (approved):
`packages/kbn-babel-preset/styled_components_files.js` LGTM

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/268240)
* Commit: 09a4ab72d1aeb30eb8ee05d337369b6b9a6f9e79

### Failed CI Steps
* [Jest Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/268240#01947649-3c48-4eaf-b6d9-70136359dc59)
* [Jest Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/268240#0194767a-b072-499c-869a-a981d8fe6f24)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/268240#01947649-3c48-4eaf-b6d9-70136359dc59) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/268240/jobs/01947649-3c48-4eaf-b6d9-70136359dc59/artifacts/0194767a-818c-4849-9f28-3eb279b5cf11) Jest Tests #<span></span>3 / ExceptionItemCardComments should render comments panel closed
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/268240#0194767a-b072-499c-869a-a981d8fe6f24) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/268240/jobs/0194767a-b072-499c-869a-a981d8fe6f24/artifacts/0194767f-4d6f-4273-8e92-69a520175eaf) Jest Tests #<span></span>3 / ExceptionItemCardComments should render comments panel closed
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/268240#01947649-3c48-4eaf-b6d9-70136359dc59) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/268240/jobs/01947649-3c48-4eaf-b6d9-70136359dc59/artifacts/0194767a-818d-4671-946c-6717285187b9) Jest Tests #<span></span>3 / ExceptionItemCardComments should render comments panel opened when accordion is clicked
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/268240#0194767a-b072-499c-869a-a981d8fe6f24) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/268240/jobs/0194767a-b072-499c-869a-a981d8fe6f24/artifacts/0194767f-4d6e-48cd-9147-e43d66402918) Jest Tests #<span></span>3 / ExceptionItemCardComments should render comments panel opened when accordion is clicked
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/268240#01947649-3c48-4eaf-b6d9-70136359dc59) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/268240/jobs/01947649-3c48-4eaf-b6d9-70136359dc59/artifacts/0194767a-818f-4b9c-b531-40930d43303f) Jest Tests #<span></span>3 / ExceptionListHeader should render edit modal
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/268240#0194767a-b072-499c-869a-a981d8fe6f24) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/268240/jobs/0194767a-b072-499c-869a-a981d8fe6f24/artifacts/0194767f-4d6d-4df2-b193-79f580f60946) Jest Tests #<span></span>3 / ExceptionListHeader should render edit modal
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/268240#01947649-3c48-4eaf-b6d9-70136359dc59) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/268240/jobs/01947649-3c48-4eaf-b6d9-70136359dc59/artifacts/0194767a-818d-4146-bd7d-9ad8274b2be7) Jest Tests #<span></span>3 / ExceptionListHeader should render the List Header with name, default description and  actions
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/268240#0194767a-b072-499c-869a-a981d8fe6f24) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/268240/jobs/0194767a-b072-499c-869a-a981d8fe6f24/artifacts/0194767f-4d6f-4d03-8bc3-8446fcd1385c) Jest Tests #<span></span>3 / ExceptionListHeader should render the List Header with name, default description and  actions
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/268240#01947649-3c48-4eaf-b6d9-70136359dc59) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/268240/jobs/01947649-3c48-4eaf-b6d9-70136359dc59/artifacts/0194767a-818f-4421-8f1e-044be8e8059f) Jest Tests #<span></span>3 / ExceptionListHeader should render the List Header with name, default description and disabled actions because of the ReadOnly mode
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/268240#0194767a-b072-499c-869a-a981d8fe6f24) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/268240/jobs/0194767a-b072-499c-869a-a981d8fe6f24/artifacts/0194767f-4d6d-47ad-bcc5-40a748606afd) Jest Tests #<span></span>3 / ExceptionListHeader should render the List Header with name, default description and disabled actions because of the ReadOnly mode
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/268240#01947649-3c48-4eaf-b6d9-70136359dc59) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/268240/jobs/01947649-3c48-4eaf-b6d9-70136359dc59/artifacts/0194767a-818d-4d9b-8a14-52b982f6df1b) Jest Tests #<span></span>3 / ExceptionListHeader should render the List Header with name, default description and disabled actions because user can not edit details
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/268240#0194767a-b072-499c-869a-a981d8fe6f24) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/268240/jobs/0194767a-b072-499c-869a-a981d8fe6f24/artifacts/0194767f-4d6f-45ae-b157-501d23af83c3) Jest Tests #<span></span>3 / ExceptionListHeader should render the List Header with name, default description and disabled actions because user can not edit details

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/268240","number":268240,"commit":"09a4ab72d1aeb30eb8ee05d337369b6b9a6f9e79"}],"number":268240}
buildkite-pr-comment-->

rylnd committed:
Fix outdated snapshots

We changed some props being passed to EUI components, which changed the
resultant HTML, here.

rylnd committed:
Merge branch 'main' into rylnd/eui_refresh_qa

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/268260)
* Commit: c79f178d5b97dfccd600e1ff1fb0c13b7c2649b4

### Failed CI Steps
* [FTR Configs #22](https://buildkite.com/elastic/kibana-pull-request/builds/268260#01947716-a7f5-4296-acdd-a480a53714eb)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/268260#01947716-a7f5-4296-acdd-a480a53714eb) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/268260/jobs/01947716-a7f5-4296-acdd-a480a53714eb/artifacts/01947736-a2fb-4926-9b55-8112f13fa2c1) FTR Configs #<span></span>22 / console app console autocomplete feature Autocomplete behavior JSON autocompletion with placeholder fields

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/4f4637da58db16cc25943793326ac6e1b85621e6) | [after](https://github.com/elastic/kibana/commit/c79f178d5b97dfccd600e1ff1fb0c13b7c2649b4) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 6562 | 6561 | -1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/4f4637da58db16cc25943793326ac6e1b85621e6) | [after](https://github.com/elastic/kibana/commit/c79f178d5b97dfccd600e1ff1fb0c13b7c2649b4) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 21.2MB | 21.2MB | -3.8KB |



### History
* :broken_heart: [Build #268240](https://buildkite.com/elastic/kibana-pull-request/builds/268240) failed 09a4ab72d1aeb30eb8ee05d337369b6b9a6f9e79

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/268260","number":268260,"commit":"c79f178d5b97dfccd600e1ff1fb0c13b7c2649b4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/268240","number":268240,"commit":"09a4ab72d1aeb30eb8ee05d337369b6b9a6f9e79"}],"number":268260}
buildkite-pr-comment-->

mgadewoll commented on x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_creation/components/eql_query_edit/eql_query_bar.tsx:
```diff
@@ -8,9 +8,9 @@
 import React, { useCallback, useEffect, useRef, useMemo } from 'react';
 import { Subscription } from 'rxjs';
-import styled from 'styled-components';
+import { css } from '@emotion/css';
```
Why is `@emotion/css` needed here? We generally advice to use `@emotion/react` with the `css` prop. ([info on the distinction](https://github.com/elastic/eui/discussions/6828#discussioncomment-6076131))

(same for other usages done with these changes)

rylnd replied:
@mgadewoll that general advice seemed, to me, to be overruled by [this comment](https://github.com/elastic/eui/discussions/6828#discussioncomment-6076131):

> If your Kibana plugin is set up with styled-components, it will not support @emotion/react and you must use @emotion/css instead

which, revisiting this based on your feedback, is perhaps too broad of a statement? Is this functionality actually at the level of an individual file, and we _can_ have both packages working in the same plugin? That is currently unclear, so I was erring on the side of the "known working" `@emotion/css`.

Additionally, the [proceeding comment](https://github.com/elastic/eui/discussions/6828#discussioncomment-6076157) about subtle bugs when mixing the two implementations does not inspire confidence that the solution is straightforward.

Please advise.

mgadewoll replied:
Thanks for the insights! That's why I asked what's the reasoning here, since I'm not having the overview across plugins. Then your update is fine. 🙂 

If you're still mixing styled-components and emotion, then there are limitations indeed. If you run into those and need a middle ground until all is migrated to emotion, there are two ways to do that:
- use `@emotion/css` as stated in the FAQ
- use `emotion/styled` as direct replacement for `styled` from `styled-components` ([FAQ](https://github.com/elastic/eui/discussions/6828#discussioncomment-6076157))

mgadewoll commented on x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_exceptions/components/exception_item_card/conditions.tsx:
```diff
@@ -86,6 +82,12 @@ export interface CriteriaConditionsProps {
+    const borderStyles = css`
+      border: ${euiTheme.border.thin};
+    `;
+    const panelStyles = cx('eui-xScroll', borderStyles);
```
As mentioned above we should use `@emotion/react` and the `css` prop instead.

For the global class you can still use the `className` prop

```
<EuiPanel
  className="eui-xScroll"
  css={borderStyles}
/>
```

rylnd replied:
Thanks! Once we have an answer on https://github.com/elastic/kibana/pull/205990#discussion_r1924272140 I will make the appropriate changes, here 👍

rylnd committed:
Merge branch 'main' into rylnd/eui_refresh_qa

 Conflicts:
	x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_creation/components/schedule_item_field/schedule_item_field.tsx

rylnd committed:
* refs/remotes/origin/rylnd/eui_refresh_qa:

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/269045)
* Commit: 82429329b00d99ef22a7020238a2be8c7207f4d5

### Failed CI Steps
* [Defend Workflows Cypress Tests on Serverless #8](https://buildkite.com/elastic/kibana-pull-request/builds/269045#01948aa6-b0cc-4d2b-83b9-7adbc4cd164c)

### History
* :yellow_heart: [Build #268260](https://buildkite.com/elastic/kibana-pull-request/builds/268260) was flaky c79f178d5b97dfccd600e1ff1fb0c13b7c2649b4
* :broken_heart: [Build #268240](https://buildkite.com/elastic/kibana-pull-request/builds/268240) failed 09a4ab72d1aeb30eb8ee05d337369b6b9a6f9e79

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/269045","number":269045,"commit":"82429329b00d99ef22a7020238a2be8c7207f4d5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/268260","number":268260,"commit":"c79f178d5b97dfccd600e1ff1fb0c13b7c2649b4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/268240","number":268240,"commit":"09a4ab72d1aeb30eb8ee05d337369b6b9a6f9e79"}],"number":269045}
buildkite-pr-comment-->

vitaliidm commented on x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_creation_ui/components/description_step/alert_suppression_label.tsx:
```diff
@@ -33,7 +33,7 @@ export const AlertSuppressionLabel = ({ label, ruleType }: AlertSuppressionLabel
+            color="danger"
             style={{ marginLeft: '8px' }}
             data-test-subj="alertSuppressionInsufficientLicensingIcon"
           />
```
<img width="983" alt="Screenshot 2025-01-22 at 11 02 22" src="https://github.com/user-attachments/assets/92421f9f-0d25-4390-9590-07d3105e7cdb" />

supression duration component in Borealis has doubled right border.
Would be great if it can be fixed

vitaliidm replied:
affects only disable state


https://github.com/user-attachments/assets/c0294bf2-8fdc-4852-b5ab-f174ee0843f5

rylnd replied:
@vitaliidm good catch, I think I'd seen this but forgotten to follow up.

Should be fixed in https://github.com/elastic/kibana/pull/205990/commits/cfa6cde37803c1d9622adbaa3ad5b9bf6fdc403f; here are some screenshots:

<details>
  <summary>
    <h3>Duration Input - Amsterdam</h3>
  </summary>

 <kbd>
 <h4>Enabled</h4>
<img width="586" alt="Duration Input - Amsterdam" src="https://github.com/user-attachments/assets/a39333c1-3057-4283-9949-3668a195814e" />
 </kbd>

 <kbd>
    <h4>Disabled</h4>
<img width="586" alt="Duration Input (Disabled) - Amsterdam" src="https://github.com/user-attachments/assets/7f775ec8-413d-4185-a0f9-1cbf894359c5" />
  </kbd>
 
</details>

<details>
  <summary>
    <h3>Duration Input - Borealis</h3>
  </summary>

 <kbd>
 <h4>Enabled</h4>

<img width="586" alt="Duration Input - Borealis" src="https://github.com/user-attachments/assets/9af7d515-2ecc-4047-b018-d1a9f5a1a345" />
 </kbd>

 <kbd>
    <h4>Disabled</h4>
<img width="586" alt="Duration Input (Disabled) - Borealis" src="https://github.com/user-attachments/assets/79093aa3-1a1b-4a7a-823e-abe0dc5ca646" />

  </kbd>
 
</details>

vitaliidm reviewed (approved):
@rylnd 

Thank you for working on this.
I left one comment that would be great to address before merging

rylnd committed:
Fix styling of Duration input when disabled

This incorrectly had a 2px right border when disabled, due to a) the
box-shadow style being overridden by inner EUI disabled styling. This
fixes that issue, and also adds a border for contrast when the input is
disabled (the light gray of the disabled numerical input was too
low-contrast sitting next to the light blue of the duration input).

rylnd committed:
Prefer @emotion/react to @emotion/css

We only need @emotion/css in files that _also_ use styled-components,
which none of these do. I previously did not understand this limitation,
and so was overly conservative with my usage of @emotion/react.

rylnd committed:
Merge branch 'main' into rylnd/eui_refresh_qa

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/269508)
* Commit: c3b8b522a75d6073bc7d488057ea078ffd9573e0

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/7f98b6bf5dc161b513abf4a85a893029e8ba5fee) | [after](https://github.com/elastic/kibana/commit/c3b8b522a75d6073bc7d488057ea078ffd9573e0) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 6609 | 6608 | -1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/7f98b6bf5dc161b513abf4a85a893029e8ba5fee) | [after](https://github.com/elastic/kibana/commit/c3b8b522a75d6073bc7d488057ea078ffd9573e0) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 21.3MB | 21.3MB | -3.5KB |



### History
* :broken_heart: [Build #269045](https://buildkite.com/elastic/kibana-pull-request/builds/269045) failed 82429329b00d99ef22a7020238a2be8c7207f4d5
* :yellow_heart: [Build #268260](https://buildkite.com/elastic/kibana-pull-request/builds/268260) was flaky c79f178d5b97dfccd600e1ff1fb0c13b7c2649b4
* :broken_heart: [Build #268240](https://buildkite.com/elastic/kibana-pull-request/builds/268240) failed 09a4ab72d1aeb30eb8ee05d337369b6b9a6f9e79

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/269508","number":269508,"commit":"c3b8b522a75d6073bc7d488057ea078ffd9573e0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/269045","number":269045,"commit":"82429329b00d99ef22a7020238a2be8c7207f4d5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/268260","number":268260,"commit":"c79f178d5b97dfccd600e1ff1fb0c13b7c2649b4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/268240","number":268240,"commit":"09a4ab72d1aeb30eb8ee05d337369b6b9a6f9e79"}],"number":269508}
buildkite-pr-comment-->
